app/main.py

```diff
@@ -2,24 +2,71 @@
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: str, name: str) -> None:
+        self.name = name
+
+    def __get__(self, instance: str, owner: str) -> None:
+        return getattr(instance, self.name)
+
+    def __set__(self, instance: str, value: str) -> None:
+        if not isinstance(value, int):
+            raise TypeError
+        if not self.min_amount <= value <= self.max_amount:
+            raise ValueError
```

Comment:
The `__set__` method should set the value in the instance's dictionary using `instance.__dict__[self.name] = value` after validation. This ensures that the descriptor correctly manages the attribute value.